### PR TITLE
Automatically create the topic and subscription.

### PR DESCRIPTION
**Pull request by SurferJeffAtGoogle**
If the first attempt at posting a message fails.

This means one fewer error that the user has to read, figure out, and fix.

### CODE REVIEW

File: pubsub/src/app.php [hunk 1]
```diff
@@ -106,4 +106,40 @@
     return new Response('', 400);
 });
 
+$app->post('/create_topic_and_subscription', function () use ($app) {
+    /** @var Google_Client $client */
+    $client = $app['google_client'];
+    $projectName = sprintf('projects/%s', $app['project_id']);
+    $pubsub = new Google_Service_Pubsub($client);
+    $topic = new \Google_Service_Pubsub_Topic();
```
Comment (bshaffer) [inline pubsub/src/app.php]: The starting backslash is not required here

**Comment by tmatsuo** [comment]
LGTM

File: pubsub/web/js/pubsub.js [hunk 2]
```diff
@@ -54,6 +55,21 @@ pubsub.PubsubController.prototype.sendMessage = function(message) {
     self.message = null;
   }).error(function(data, status) {
     self.logger.error('Failed to send the message. Status: ' + status + '.');
+    if (!self.createdTopicAndSubscription) {
```
Comment (bshaffer) [inline pubsub/web/js/pubsub.js]: Shouldn't this be `hasAttemptedToCreateTopicAndSubscription`?

File: pubsub/src/app.php (hunk 1 above)
Comment (SurferJeffAtGoogle) [inline pubsub/src/app.php]: Done.

File: pubsub/web/js/pubsub.js (hunk 2 above)
Comment (SurferJeffAtGoogle) [inline pubsub/web/js/pubsub.js]: Done.

**Pull request merged**